libvast/vast/format/pcap.hpp

```diff
@@ -39,6 +41,22 @@ namespace vast {
 namespace format {
 namespace pcap {
 
+struct pcap_deleter_t {
+  template <class THandle>
+  auto operator()(THandle handle) {
+    if (handle)
+      ::pcap_close(handle);
+  }
+};
```

Comment:
```suggestion
struct pcap_close_wrapper {
  inline void operator()(pcap_t* handle) const noexcept {
    ::pcap_close(handle);
  }
};
```

The deleter of a `unique_ptr` is not called when the underlying pointer is `nullptr`. No need to double check.

I think it makes more sense to name the struct `pcap_close_wrapper`, similar to the `actor_cast_wrapper` we already have in our code base. There's also no need to use a template here.

Please apply the same change to the other deleter.

Reply:
Good call on not needing the template. Just adopted your proposed changes, take a look.

Reply:
You're missing the `inline`. Alternatively you can also move the definition to the source files, which might be a good idea in general so we can make libpcap a private dependency sometime in the future.

Reply:
Added the `inline` keyword. I don't want to go down moving the definition to the `cpp` file as the deleter needs to be a complete definition when declaring the `unique_ptr` data members.

Reply:
I might just do this myself in the future. We want to externalize the PCAP reader/writer into a plugin anyways so we don't always depend on libpcap, and it's only natural to make the change then.

Reply:
Seems like a reasonable idea; feel free.